library/core/src/panicking.rs

```diff
@@ -60,6 +60,13 @@ pub fn panic_str(expr: &str) -> ! {
     panic_fmt(format_args!("{}", expr));
 }
 
+#[inline]
+#[track_caller]
+#[cfg_attr(not(bootstrap), lang = "panic_display")] // needed for const-evaluated panics
+pub fn panic_display<T: fmt::Display>(x: &T) -> ! {
```

Comment:
Looks like every call to `panic_str` could be replaced by a call to `panic_display`... why do we need both functions?

Reply:
We should bench that first, it may cause compile-time regressions, but if not, yea, we could just replace panic_str

Reply:
That'll allow no_std code to do `panic!(1)`. I don't think that's desirable...

Reply:
Oh of course, explicit args, how did I forget that :D

Reply:
I had a quick look and it turns out `panic_str` is needed because `non_fmt_panic` lint requires that to tell the difference between `panic!(STR)` and `panic!("{}", STR)` and generate a future compat lint for the former.

Reply:
In that case pleas add a comment all the panic_str call site explaining this,.

Reply:
Added